lua/entities/gmod_wire_expression2/core/custom/cfc_vector2_NumberOperands.lua

```diff
@@ -0,0 +1,537 @@
+    end
+    return {
+        intNum(this[1]),
+        intNum(this[2])
+    }
+end
+
+--- returns the fractional part. (frac(-1.5) == 0.5 & frac(3.2) == 0.2)
+e2function vector2 frac(vector2 vectah)
+    local function fracnum(number)
+        if rv1 >= 0 then return rv1 % 1 else return rv1 % -1 end
+    end
+    return {
+        fracnum(vectah[1]),
+        fracnum(vectah[2])
+    }
+end
+
+e2function vector2 vector2:frac()
+    local function fracnum(number)
+        if rv1 >= 0 then return rv1 % 1 else return rv1 % -1 end
+    end
+    return {
+        fracnum(this[1]),
+        fracnum(this[2])
+    }
+end
+
+e2function vector2 vector2:wrap(vector2 wrapby)
+    return {
+        (this[1] + wrapby[1]) % (wrapby[1] * 2) - wrapby[1],
+        (this[2] + wrapby[2]) % (wrapby[2] * 2) - wrapby[2]
+    }
+end
+
+e2function vector2 wrap(vector2 wrapped, vector2 wrapby)
+    return {
+        (wrapped[1] + wrapby[1]) % (wrapby[1] * 2) - wrapby[1],
+        (wrapped[2] + wrapby[2]) % (wrapby[2] * 2) - wrapby[2]
+    }
+end
+
+--[[************************************************************************]]--
+
+__e2setcost(2) -- approximation
+
+e2function vector2 sqrt(vector2 vectah)
+    return {
+        vectah[1] ^ (1 / 2),
+        vectah[2] ^ (1 / 2)
+    }
+end
+e2function vector2 vector2:sqrt()
+    return {
+        this[1] ^ (1 / 2),
+        this[2] ^ (1 / 2)
+    }
+end
+
+e2function vector2 cbrt(vector2 vectah)
+    return {
+        vectah[1] ^ (1 / 3),
+        vectah[2] ^ (1 / 3)
+    }
+end
+e2function vector2 vector2:cbrt()
+    return {
+        this[1] ^ (1 / 3),
+        this[2] ^ (1 / 3)
+    }
+end
+
+e2function vector2 root(vector2 vectah, number num)
+    return {
+        vectah[1] ^ (1 / num),
+        vectah[2] ^ (1 / num)
+    }
+end
+e2function vector2 vector2:root(number num)
+    return {
+        this[1] ^ (1 / num),
+        this[2] ^ (1 / num)
+    }
+end
+
+e2function vector2 root2(vector2 vectah, vector2 vecta)
+    return {
+        vectah[1] ^ (1 / vecta[1]),
+        vectah[2] ^ (1 / vecta[2])
+    }
+end
+e2function vector2 vector2:root2(vector2 vecta)
+    return {
+        this[1] ^ (1 / vecta[1]),
+        this[2] ^ (1 / vecta[2])
+    }
+end
+
+--[[************************************************************************]]--
+
+__e2setcost(2) -- approximation
+
+local deg2rad = pi / 180
+local rad2deg = 180 / pi
+
+e2function vector2 toRad(vector2 vectah)
+    return {
+        vectah[1] * deg2rad,
+        vectah[2] * deg2rad
+    }
+end
+e2function vector2 vector2:toRad()
+    return {
+        this[1] * deg2rad,
+        this[2] * deg2rad
+    }
+end
+
+e2function vector2 toDeg(vector2 vectah)
+    return {
+        vectah[1] * rad2deg,
+        vectah[2] * rad2deg
+    }
+end
+e2function vector2 vector2:toDeg()
+    return {
+        this[1] * rad2deg,
+        this[2] * rad2deg
+    }
+end
+
+e2function vector2 acosvec2(vector2 vectah)
```

Comment:
```suggestion
e2function vector2 acos(vector2 vectah)
```

Not really needed? Easier to remember

Reply:
It is needed. I would suggest you to read the function right after it.